### PR TITLE
fix issues affecting code quality

### PR DESCRIPTION
**Pull request by withshubh**
#### Description
Hey, I work at DeepSource, I ran DeepSource analysis on my fork of the repo and found [some improvements](https://deepsource.io/gh/withshubh/nettacker/issues/?category=recommended) in the codebase, opening this PR so you can assess if our platform is right and helpful for you.

#### Changes proposed in this pull request
- Removed length check in favour of truthiness of the object
> boosts minor performance, know more [here](https://deepsource.io/gh/withshubh/Nettacker/issue/PYL-C1801/description)
- Used identity check for comparison with True
> boosts minor performance, know more [here](https://deepsource.io/gh/withshubh/Nettacker/issue/FLK-E712/description/)
- Removed unnecessary use of comprehension
> boosts minor performance and improves readability, know more [here](https://deepsource.io/gh/withshubh/Nettacker/issue/PYL-R1721/description/)
- Used negative index `-1` to get the last element of sequence
> boosts minor performance, know more [here](https://deepsource.io/gh/withshubh/Nettacker/issue/PTC-W0044/description/)
- Used `callable()` to check if the object is callable
> potential bug risk, know more [here](https://deepsource.io/gh/withshubh/Nettacker/issue/PTC-W0035/description/)
- Added .deepsource.toml
> configuration file to continuously analyze the repo for code quality issues

#### Checklist
- [x] I have followed the [Contributor Guidelines](https://github.com/OWASP/Nettacker/wiki/Developers#contribution-guidelines).
- [x] The code is both Python 2 and Python 3 compatible.
- [x] The code follows the PEP8 styling guidelines with 4 spaces indentation.
- [x] This Pull Request relates to only one issue or only one feature
- [x] My branch is up-to-date with the Upstream master branch.

### CODE REVIEW

**Comment by securestep9** [comment]
hey Shubhendra @withshubh - many thanks for this. Wee are aware of multiple code quality issues. Will review your PR, just give us some time please.

File: .deepsource.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version = 1
+
+exclude_patterns = [
```
Comment (itsdivyanshjain) [inline .deepsource.toml:3]: we should exclude database directory as well because it generally requires no modification

File: .deepsource.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    "scripts/**",
+    "web/**"
+]
+
+[[analyzers]]
```
Comment (itsdivyanshjain) [inline .deepsource.toml:8]: There is docker analyzer we may need to add that

**Comment by itsdivyanshjain** [comment]
@withshubh  @securestep9 

Deepsource.io is pretty work well to detect `anti-pattern`, `bug risk` as `security-bug` and helps us solve it but there auto-fix feature gives sometimes vague results.

**Comment by withshubh** [comment]
Hi @securestep9 :wave: 

Please review this PR! :sparkles:

**Comment by Ali-Razmjoo** [comment]
fixed #440

**Pull request closed without merging**